modules/artifact/src/main/java/org/mule/runtime/module/artifact/descriptor/BundleDependency.java

```diff
@@ -0,0 +1,163 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.module.artifact.descriptor;
+
+import static com.google.common.base.Preconditions.checkState;
+import static java.lang.String.format;
+import static java.util.Optional.empty;
+import static java.util.Optional.of;
+
+import java.util.Optional;
+
+/**
+ * Describes a dependency on a bundle.
+ *
+ * @since 4.0
+ */
+public class BundleDependency {
+
+  private BundleDescriptor descriptor;
+  private String type;
+  private Optional<String> classifier = empty();
```

Comment:
shouldn't both the type and classifier be part of the BundleDescriptor?. If BundleDescriptor is the one that uniquely identifies the dependency then it should also have type and classifier. The I guess the BundleDependency declares a particular dependency to a bundle descriptor and that's why it has the scope attribute